src/stream-analytics.ts

```diff
@@ -65,9 +77,15 @@ class StreamAnalytics<UserType = unknown> {
         return this._sendEvent('impression', eventData);
     }
 
-    trackEngagement(eventData: Engagement) {
-        const err = validateEngagement(eventData);
-        if (err) throw new errors.InvalidInputData('event data is not valid', err);
+    trackEngagement(...eventData: Engagement[]) {
```

Comment:
IMO we should go with array param instead of rest parameters, we can overload the function `trackEngagement({})` and `trackEngagements([{},{}])`